src/validator.ts

```diff
@@ -680,6 +682,51 @@ function doIntrinsicGetAZs(ref: any, key: string){
 
 }
 
+function doIntrinsicSelect(ref: any, key: string){
+  let toGet = ref[key];
+  if(!Array.isArray(toGet) || toGet.length < 2) {
+    addError('crit', "Fn::Select only supports an array of two elements", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[0] === undefined || toGet[0] === null) {
+     addError('crit', "Fn::Select first element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let index = parseInt(toGet[0]);
+  if (typeof index !== 'number' || isNaN(index)) {
+    addError('crit', "First element of Fn::Select must be a number", placeInTemplate, "Fn::Select");
+    return 'INVALID_SELECT';
+  }
+  if (toGet[1] === undefined || toGet[1] === null) {
+     addError('crit', "Fn::Select Second element cannot be null or undefined", placeInTemplate, "Fn::Select");
+     return 'INVALID_SELECT';
+  }
+
+  let list = toGet[1]
+  if (!Array.isArray(list)) {
+    //we may need to resolve it
+    if (typeof list !== 'string') {
+      list = resolveIntrinsicFunction(list, Object.keys(list)[0]);
+    } else {
+      addError('crit', "Fn::Select requires the second element to resolve to a list, it appears to be a string", placeInTemplate, "Fn::Select");
+      return 'INVALID_SELECT';
+    }
+    if (!Array.isArray(list)) {
```

Comment:
As we are expecting an Array, can we check for the following property of null entries?
> This list must not be null, nor can it have null entries.

Reply:
Yep.